docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,425 @@
+
+
+### Using kubeadm init with a configuration file {#config-file}
+
+**Caution:** the config file is
+still considered alpha and may change in future versions.
+{: .caution}
+
+It's possible to configure `kubeadm init` with a configuration file instead of command
+line flags, and some more advanced features may only be available as
+configuration file options.  This file is passed in to the `--config` option.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+api:
+  advertiseAddress: <address|string>
+  bindPort: <int>
+etcd:
+  endpoints:
+  - <endpoint1|string>
+  - <endpoint2|string>
+  caFile: <path|string>
+  certFile: <path|string>
+  keyFile: <path|string>
+  dataDir: <path|string>
+  extraArgs:
+    <argument>: <value|string>
+    <argument>: <value|string>
+  image: <string>
+networking:
+  dnsDomain: <string>
+  serviceSubnet: <cidr>
+  podSubnet: <cidr>
+kubernetesVersion: <string>
+cloudProvider: <string>
+nodeName: <string>
+authorizationModes:
+- <authorizationMode1|string>
+- <authorizationMode2|string>
+token: <string>
+tokenTTL: <time duration>
+selfHosted: <bool>
+apiServerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+controllerManagerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+schedulerExtraArgs:
+  <argument>: <value|string>
+  <argument>: <value|string>
+apiServerCertSANs:
+- <name1|string>
+- <name2|string>
+certificatesDir: <string>
+imageRepository: <string>
+unifiedControlPlaneImage: <string>
+featureGates:
+  <feature>: <bool>
+  <feature>: <bool>
+```
+
+### Passing custom arguments to control plane components {#custom-args}
+
+If you would like to override or extend the behaviour of a control plane component, you can provide 
+extra arguments to kubeadm. When the component is deployed, it will use these additional arguments in 
+the pod command itself. 
+
+For example, to add additional feature-gate arguments to the apiserver, your [configuration file](#sample-master-configuration) 
+will need to look like this:
+
+```
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+apiServerExtraArgs:
+   feature-gates: APIResponseCompression=true
+```
+
+To customise the scheduler or controller-manager, use `schedulerExtraArgs` and `controllerManagerExtraArgs` respectively.
+
+More information on custom arguments can be found here:
+- [kube-apiserver](https://kubernetes.io/docs/admin/kube-apiserver/)
+- [kube-controller-manager](https://kubernetes.io/docs/admin/kube-controller-manager/)
+- [kube-scheduler](https://kubernetes.io/docs/admin/kube-scheduler/)
+
+### Using custom images {#custom-images}
+
+By default, kubeadm will pull images from `gcr.io/google_containers`, unless
+the requested Kubernetes version is a CI version. In this case,
+`gcr.io/kubernetes-ci-image` will be used.
+
+This behaviour can be overridden by [using kubeadm with a configuration file](#config-file).
+Allowed customization are:
+
```

Comment:
Global comment: Use active voice whenever possible. Here, for example, you could say, " You can override this behavior by ..."